interpreter/src/main/scala/pl/touk/nussknacker/engine/language/dictWithLabel/DictKeyWithLabelExpressionParser.scala

```diff
@@ -54,10 +53,20 @@ object DictKeyWithLabelExpressionParser extends ExpressionParser {
         }
     }
 
-  case class CompiledDictKeyExpression(key: String) extends CompiledExpression {
+  case class CompiledDictKeyExpression(key: String, expectedType: TypingResult) extends CompiledExpression {
     override def language: Language = languageId
 
-    override def evaluate[T](ctx: Context, globals: Map[String, Any]): T = key.asInstanceOf[T]
+    override def evaluate[T](ctx: Context, globals: Map[String, Any]): T = {
+      if (expectedType.canBeSubclassOf(Typed[Long])) {
+        key.toLong.asInstanceOf[T]
+      } else if (expectedType.canBeSubclassOf(Typed[Boolean])) {
+        key.toBoolean.asInstanceOf[T]
+      } else if (expectedType.canBeSubclassOf(Typed[String])) {
+        key.asInstanceOf[T]
+      } else {
+        throw new IllegalStateException(s"DictKeyExpression of unsupported type: ${key.getClass}")
```

Comment:
`key.getClass` can be inlined to String :) I think that you wanted to say that this value can't be converted to expectedType which we don't know during compilation. Am I right?